### ISSUE TITLE
Dataset size decreasing over time

### ISSUE THREAD

**Issue by gwillem** [rapid7/sonar#4]
Hi Rapid7, thanks for sharing these awesome datasets!

Am looking at the [DNS-ANY](https://scans.io/study/sonar.fdns) sets and it seems that the size decreased over time, while I would rather expect them to grow. 

Would you have any explanation for that?

**Comment by jhart-r7** [comment]
In general I agree -- most of these datasets should grow over time, not shrink.  But, as to the forward DNS study, it is unfortunately somewhat expected that as of the last ~month its size may have dropped a bit.  

The current process we use to perform both the forward and reverse DNS study has long been a source of abuse complaints and recently we had to change things up a bit in order to lessen the abuse.  In particular we have reduced the number of retries used to attempt to elicit a valid, non-error response from the downstream authoritative nameserver, but unfortunately this means that if we encounter any given nameserver that is authoritative for disproportionally more records than it can reasonably handle, we may end up missing some records.  

All of this is under active development and will likely change more in the future.  In particular, the FDNS study may abandon doing regular ANY queries because they are routinely abused, used as DDoS vectors, have IDS/IPS signatures, and in general cause quite a bit of noise.  Obviously the data is useful so we are looking for a middle ground that keeps everyone happy enough.

Thanks!
